projects/core/src/i18n/translate.pipe.spec.ts

```diff
@@ -1,25 +1,36 @@
-import { ChangeDetectorRef } from '@angular/core';
+import { ChangeDetectorRef, Injectable } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
 import { EMPTY, of } from 'rxjs';
+import { MockTranslationService } from './testing/mock-translation.service';
 import { TranslatePipe } from './translate.pipe';
 import { TranslationService } from './translation.service';
-import createSpy = jasmine.createSpy;
+
+@Injectable()
+class MockChangeDetectorRef implements Partial<ChangeDetectorRef> {
+  markForCheck() {}
+}
 
 describe('TranslatePipe', () => {
   let pipe: TranslatePipe;
   let service: TranslationService;
   let cd: ChangeDetectorRef;
 
   beforeEach(() => {
-    service = {
-      translate: () => {},
-    } as any;
-    cd = { markForCheck: createSpy('markForCheck') } as any;
-    pipe = new TranslatePipe(service, cd);
+    TestBed.configureTestingModule({
+      providers: [
+        TranslatePipe,
+        { provide: TranslationService, useClass: MockTranslationService },
+        { provide: ChangeDetectorRef, useClass: MockChangeDetectorRef },
+      ],
+    });
+    pipe = TestBed.inject(TranslatePipe);
+    service = TestBed.inject(TranslationService);
+    cd = TestBed.inject(ChangeDetectorRef);
   });
 
   describe('transform', () => {
     it('should return falsy when input is undefined', () => {
-      const result = pipe.transform(undefined);
+      const result = pipe.transform('');
```

Comment:
[QUESTION]

Are changes in this file _really_ related to this PR?

btw. why we changed `undefined` to `''`? The description says `should return falsy when input is undefined`

Reply:
No, this particular change is not related to this PR. I'll redo it.
Btw, I would change `undefined` to `''` and the test description too, because the definition of this function does not allow to provide `undefined`:
```
transform(
    input: Translatable | string,
    options: TranslatableParams = {}
  )
```